src/functions/retryProcessing/application/__tests__/AutosaveErrorsToRetry.int.ts

```diff
@@ -18,23 +36,228 @@ describe('AutoSaveErrorsToRetry', () => {
+      });
+
+    expect(await getAutosaveQueueRecord(db, InterfaceIds.NOTIFY,  ErrorsToRetryTestCases.TarsProcessingRsisFailed))
+    .toContain(
+      { application_reference: ErrorsToRetryTestCases.TarsProcessingRsisFailed,
+        interface: InterfaceIds.NOTIFY,
+        upload_status: ProcessingStatus.PROCESSING,
+      });
+
+    expect(await getAutosaveQueueRecord(db, InterfaceIds.TARS,  ErrorsToRetryTestCases.TarsAcceptedRsisFailed))
+    .toContain(
+      { application_reference: ErrorsToRetryTestCases.TarsAcceptedRsisFailed,
+        interface: InterfaceIds.TARS,
+        upload_status: ProcessingStatus.ACCEPTED,
+      });
+
+    expect(await getAutosaveQueueRecord(db, InterfaceIds.NOTIFY,  ErrorsToRetryTestCases.TarsAcceptedRsisFailed))
+    .toContain(
+      { application_reference: ErrorsToRetryTestCases.TarsAcceptedRsisFailed,
+        interface: InterfaceIds.NOTIFY,
+        upload_status: ProcessingStatus.PROCESSING });
+
+    // ensure autosave flag is still set
+    expect(await getTestResultAutosaveFlag(db, ErrorsToRetryTestCases.TarsFailedRsisFailed)).toBe(1);
+    expect(await getTestResultAutosaveFlag(db, ErrorsToRetryTestCases.TarsFailedRsisProcessing)).toBe(1);
+    expect(await getTestResultAutosaveFlag(db, ErrorsToRetryTestCases.TarsFailedRsisAccepted)).toBe(1);
+    expect(await getTestResultAutosaveFlag(db, ErrorsToRetryTestCases.TarsProcessingRsisFailed)).toBe(1);
+    expect(await getTestResultAutosaveFlag(db, ErrorsToRetryTestCases.TarsAcceptedRsisFailed)).toBe(1);
+
+      // ensure no RSIS entries created
+    expect(await getQueueCount(db,
+                               ErrorsToRetryTestCases.TarsFailedRsisFailed,
+                               InterfaceIds.RSIS)).toBe(0);
+    expect(await getQueueCount(db,
+                               ErrorsToRetryTestCases.TarsFailedRsisProcessing,
+                               InterfaceIds.RSIS)).toBe(0);
+    expect(await getQueueCount(db,
+                               ErrorsToRetryTestCases.TarsFailedRsisAccepted,
+                               InterfaceIds.RSIS)).toBe(0);
+    expect(await getQueueCount(db,
+                               ErrorsToRetryTestCases.TarsProcessingRsisFailed,
+                               InterfaceIds.RSIS)).toBe(0);
+    expect(await getQueueCount(db,
+                               ErrorsToRetryTestCases.TarsAcceptedRsisFailed,
+                               InterfaceIds.RSIS)).toBe(0);
 
-    expect(autosaveRecords).toContain({ application_reference: 69, interface: 0, upload_status: 1 });
-    expect(autosaveRecords).toContain({ application_reference: 69, interface: 1, upload_status: 0 });
   });
 });
+
+const getTestResultData = (): AutosaveTestData[] => {
+  return [
+    {
+      applicationReference: ErrorsToRetryTestCases.TarsFailedRsisFailed,
+      staffNumber: '1',
+      driverSurname: 'Bloggs',
+      resultStatus: ProcessingStatus.PROCESSING,
+      autosave: true,
+    },
```

Comment:
Could be refactored into a function, up to you if you think it's worth it:

`function createRecord(appRef: number, staffNum: string, interf: number, uploadStat: number, retryCou: number) {
  return {
    applicationReference: appRef,
    staffNumber: staffNum,
    interface: interf,
    uploadStatus: uploadStat,
    retryCount: retryCou,
  };
}

createRecord(ErrorsToRetryTestCases.TarsFailedRsisFailed, '1', InterfaceIds.TARS, ProcessingStatus.FAILED, 0);`